packages/cli/src/commands/build/index.ts

```diff
@@ -826,6 +832,43 @@ function mergeFlags(
   });
 }
 
+/**
+ * Takes the build output and writes all the variants into the `variants.json`
+ * file. It'll skip variants that already exist.
+ */
+async function mergeVariants(
+  { output }: Client,
+  buildResults: Iterable<BuildResult | BuildOutputConfig>,
+  outputDir: string
+): Promise<void> {
+  const variantsFilePath = join(outputDir, 'variants.json');
+
+  const variants = (await fs.readJSON(variantsFilePath).catch(error => {
+    if (error.code === 'ENOENT') {
+      return { defintions: {} };
+    }
+
+    throw error;
+  })) as { definitions: Record<string, VariantDefinition> };
+
+  for (const result of buildResults) {
+    if (!('variants' in result) || !result.variants) continue;
+
+    for (const [key, defintion] of Object.entries(result.variants)) {
+      if (result.variants[key]) {
+        output.warn(
+          `The variant "${key}" was found multiple times. Only its first occurrence will be considered.`
+        );
+        continue;
+      }
+
+      result.variants[key] = defintion;
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (blocking):** should this be saving to `variants` for writing later?‏

```suggestion
      variants[key] = defintion;
```

Reply:
Oh, thanks. That was an oversight.
Will add tests for this in upcoming PRs.